### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/AddProductAdminController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductAdminController extends HttpServlet {
```
Comment (DmGrt) [inline src/main/java/com/internet/shop/controllers/AddProductAdminController.java]: We can call this class as AddProductController

File: src/main/java/com/internet/shop/controllers/DeleteOrderAdminController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteOrderAdminController extends HttpServlet {
```
Comment (DmGrt) [inline src/main/java/com/internet/shop/controllers/DeleteOrderAdminController.java]: Here we can remove the admin to

File: src/main/java/com/internet/shop/controllers/GetAllOrdersAdminController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllOrdersAdminController extends HttpServlet {
```
Comment (DmGrt) [inline src/main/java/com/internet/shop/controllers/GetAllOrdersAdminController.java]: Without admin

File: src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllProductsAdminController extends HttpServlet {
```
Comment (DmGrt) [inline src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java]: same

**Review by DmGrt — COMMENTED** [review]
Good job! Just need some changes in naming I think)

File: src/main/java/com/internet/shop/controllers/AddProductAdminController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    private static final Injector injector = Injector
+            .getInstance("com.internet.shop");
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddProductAdminController.java]: ця строка влазить в 100 символів, тому краще не ділити її на 2 рядки, аналогічно з рештою

File: src/main/java/com/internet/shop/controllers/AddProductAdminController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    private ProductService productService
+            = (ProductService) injector
+            .getInstance(ProductService.class);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/AddProductAdminController.java]: це також занадто, давай краще на 2 рядки поділяти:
```
private ProductService productService = 
        (ProductService) injector.getInstance(ProductService.class);
```

File: src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java (hunk 4 above)
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java]: тут можна назвати наприклад: ManageAllProductsController

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -93,4 +93,97 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: занадто пустих строк

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -93,4 +93,97 @@
...
+    <servlet>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetAllProductsAdminController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <url-pattern>/products/all-admin</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

File: src/main/webapp/web.xml [hunk 9]
```diff
@@ -93,4 +93,97 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allOrdersAdmin</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetAllOrdersAdminController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allOrdersAdmin</servlet-name>
+        <url-pattern>/orders/all-admin</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/all</url-pattern>
```

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -93,4 +93,97 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allOrders</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GetAllUserOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allOrders</servlet-name>
+        <url-pattern>/orders/all</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/all/by-user (or /orders/by-user)</url-pattern>
```

**Review by Sofasmile — COMMENTED** [review]
I agree with all @DmGrt comments

File: src/main/webapp/web.xml (hunk 7 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: done

File: src/main/java/com/internet/shop/controllers/AddProductAdminController.java (hunk 6 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/AddProductAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/AddProductAdminController.java (hunk 5 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/AddProductAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/AddProductAdminController.java (hunk 1 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/AddProductAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/DeleteOrderAdminController.java (hunk 2 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/DeleteOrderAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/GetAllOrdersAdminController.java (hunk 3 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/GetAllOrdersAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java (hunk 4 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java:13]: I have GetAllProductsController (whre user can see all products amd buy it) already.
so i cant rename it.

Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/GetAllProductsAdminController.java]: done

File: src/main/webapp/web.xml (hunk 8 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 10 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/web.xml (hunk 9 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: не поняла. написала в слаке

Comment (IrynaDim) [reply src/main/webapp/web.xml]: done. change to "manage" Sofia approved.

File: src/main/webapp/WEB-INF/views/orders/manage.jsp [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        <tr>
+            <td>
+                <c:out value="${order.orderId}"/>
+            </td>
+            <td>
+                <c:out value="${order.userId}"/>
+            </td>
+            <td>
+                <a href="${pageContext.request.contextPath}/orders/delete?id=${order.orderId}">Delete</a>
+            </td>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/orders/manage.jsp:25]: we may also suggest to show details here

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -93,4 +93,83 @@
...
     </servlet-mapping>
 
+    <servlet>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <servlet-class>com.internet.shop.controllers.ManageAllProductsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allProductsAdmin</servlet-name>
+        <url-pattern>/products/all-admin</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -93,4 +93,83 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>orderDetails</servlet-name>
+        <servlet-class>com.internet.shop.controllers.GerOrderDetailController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>orderDetails</servlet-name>
+        <url-pattern>/orders/by-user/details</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/details</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/webapp/web.xml (hunk 12 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: +

File: src/main/webapp/web.xml (hunk 13 above)
Comment (IrynaDim) [reply src/main/webapp/web.xml]: done

File: src/main/webapp/WEB-INF/views/orders/manage.jsp (hunk 11 above)
Comment (IrynaDim) [reply src/main/webapp/WEB-INF/views/orders/manage.jsp:25]: done

File: src/main/java/com/internet/shop/exceptions/AuthenticationException.java [hunk 14]
```diff
@@ -0,0 +1,7 @@
+package com.internet.shop.exceptions;
+
+public class AuthenticationException extends Exception {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/exceptions/AuthenticationException.java:3]: it is redundant here

File: src/main/webapp/WEB-INF/views/orders/order-details.jsp [hunk 15]
```diff
@@ -27,6 +27,6 @@
         </tr>
     </c:forEach>
 </table>
-<a href="${pageContext.request.contextPath}/orders/by-user"> Go to all your orders </a>
+<a href="${pageContext.request.contextPath}/orders/details"> Go to all your orders </a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/orders/order-details.jsp:30]:
```suggestion
<a href="${pageContext.request.contextPath}/orders/by-user"> Go to all your orders </a>
```

File: src/main/java/com/internet/shop/exceptions/AuthenticationException.java (hunk 14 above)
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/exceptions/AuthenticationException.java:3]: this is redundant here, please remove

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**